code/datums/station_traits/neutral_traits.dm

```diff
@@ -75,3 +75,77 @@
 /datum/station_trait/announcement_medbot/New()
 	. = ..()
 	SSstation.announcer = /datum/centcom_announcer/medbot
+
+/datum/station_trait/protagonist
+	name = "Announcement \"System\""
+	trait_type = STATION_TRAIT_NEUTRAL
+	weight = 0
+	show_in_report = TRUE
+	trait_flags = STATION_TRAIT_ABSTRACT
+	report_message = "This station has received an esteemed guest! They will most likely be a high value target for any Syndicate invasions. Be sure to keep them safe!"
+	blacklist = list()
+	///What role to give to the picked player
+	var/datum/antagonist/role_to_give
+	//The mind of the player that was picked
+	var/datum/mind/picked_mind
+	//The antag datum instance for the protagonist we are creating
+	var/datum/antagonist/antag_datum_instance
+
+
+/datum/station_trait/protagonist/New()
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_PRE_GAMEMODE_SETUP, .proc/on_gamemode_setup)
+	RegisterSignal(SSdcs, COMSIG_GLOB_POST_GAMEMODE_SETUP, .proc/after_gamemode_setup)
+	antag_datum_instance = new role_to_give() //Create this early for things such as family name to be setup
+
+/// Checks if candidates are connected and if they are banned or don't want to be the antagonist.
+/datum/station_trait/protagonist/proc/trim_candidates(list/candidates)
+	for(var/mob/dead/new_player/candidate_player in candidates)
+		var/client/candidate_client = GET_CLIENT(candidate_player)
+		if (!candidate_client || !candidate_player.mind) // Are they connected?
+			candidates.Remove(candidate_player)
+			continue
+
+		if(candidate_player.mind.special_role) // No double antags!
+			candidates.Remove(candidate_player)
+			continue
+
+		if(!(ROLE_PROTAGONIST in candidate_client.prefs.be_special) || is_banned_from(candidate_player.ckey, list(ROLE_PROTAGONIST, ROLE_SYNDICATE)))
+			candidates.Remove(candidate_player)
+			continue
+
+/datum/station_trait/protagonist/proc/on_gamemode_setup(datum/gamemode/gamemode_ref)
+	SIGNAL_HANDLER
+	var/list/candidates = list()
+	for(var/i in GLOB.new_player_list)
```

Comment:
instead of
``` dm
for(var/i in GLOB.new_player_list)
    var/mob/dead/new_player/player = i
```
you should use
``` dm
for(var/mob/dead/new_player/player as anything in GLOB.new_player_list)
```
less boilerplate, and it avoids the use of var/i which we want more descriptive var names

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#avoid-unnecessary-type-checks-and-obscuring-nulls-in-lists
and
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names